src/test_simple_graph.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+"""File tests our simple graph data structure."""
+from __future__ import unicode_literals
+import pytest
+
+TEST_SG_DATA = [
+    ("node_a", [])
+]
+
+
+def test_sg_init():
+    """Ensure init returns a valid dictionary at self.graph."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    assert test_sg.graph == {}
+
+
+def test_sg_nodes():
+    """Ensure nodes returns a correct list of nodes in graph."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_node = test_sg.nodes()
+    assert test_node == []
+
+
+def test_sg_edges_empty_graph():
+    """Ensure edges returns a correct list of edges in graph."""
+    from simple_graph import SimpleGraph
+    test_edge = SimpleGraph().edges()
+    assert test_edge == []
+
+
+def test_sg_edges():
+    """Ensure edges returns correct list of edges in graph."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_sg.add_edge("node_a", "node_b")
+    assert test_sg.edges() == [("node_a", "node_b")]
+
+
+@pytest.mark.parametrize('data, result', TEST_SG_DATA)
+def test_sg_add_node(data, result):
+    """Ensure new node is added to the graph."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_sg.add_node(data)
+    assert test_sg.graph[data] == result
+
+
+def test_sg_add_node_error():
+    """Ensure duplicate node raises IndexError."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_sg.add_node("node_a")
+    with pytest.raises(IndexError) as message:
+        test_sg.add_node("node_a")
+    assert "That node already exists dumbo." in str(message)
+
+
+def test_sg_add_edge():
+    """Ensure new edge is added to the graph."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_sg.add_edge("node_a", "node_b")
+    assert test_sg.graph["node_a"] == [("node_b", 1)]
+
+
+def test_sg_del_node():
+    """Ensure correct node is deleted, and all its edges are removed."""
+    from simple_graph import SimpleGraph
+    test_sg = SimpleGraph()
+    test_sg.add_edge("a", "c")
+    test_sg.add_edge("b", "c")
+    test_sg.add_edge("c", "b")
+    test_sg.del_node("c")
+    assert "c" not in test_sg.edges()
```

Comment:
This testing isn't doing what you think it is, because your `.edges()` is returning a list of tuples. the `in` operator does not check recursively inside any containers within the outer container. So you're not checking if `"c"` is in any of the tuples.

You could use this:
```
from itertools import chain
...
assert "c" not in list(chain(test_sg.edges()))
```
`itertools.chain` will "flatten" a container of containers into one container.